logzero/__init__.py

```diff
@@ -431,6 +426,40 @@ def logfile(filename, formatter=None, mode='a', maxBytes=0, backupCount=0, encod
         logger.addHandler(rotating_filehandler)
 
 
+def __remove_internal_loggers(logger_to_update, disableStderrLogger=True):
+    """
+    Remove the internal loggers (e.g. stderr logger and file logger) from the specific logger
+    :param logger_to_update: the logger to remove internal loggers from
+    :param disableStderrLogger: should the default stderr logger be disabled? defaults to True
+    """
+    for handler in list(logger_to_update.handlers):
+        if hasattr(handler, LOGZERO_INTERNAL_LOGGER_ATTR):
+            if isinstance(handler, RotatingFileHandler):
+                logger_to_update.removeHandler(handler)
+            elif isinstance(handler, SysLogHandler):
+                logger_to_update.removeHandler(handler)
+            elif isinstance(handler, logging.StreamHandler) and disableStderrLogger:
+                logger_to_update.removeHandler(handler)
+
+
+def syslog(logger_to_update=logger, facility=SysLogHandler.LOG_USER, disableStderrLogger=True):
```

Comment:
I'd rather have `disableStderrLogger` default to False, as with the `logfile` setup. What do you think?

Reply:
I think you can argue that this method shouldn't have a `disableStderrLogger` argument at all and should just remove every internal logger, as the name implies. It's purely an internal method, anyway. I left the param just for consistency, but I do think it makes sense that the default behavior is to remove every internal logger, as the name implies.

To be honest, I don't really care for the way that `disableStderrLogger` is passed all around (and I'm the one that added it 😂 ). It's a reflection of the fact that we need a `Logger` class that carries these properties internally. I _almost_ went down the path of doing that redesign yesterday but opted for the quicker implementation for now.

Given that this is purely an internal, private method, I'm inclined to leave it pending the OO-redesign, but if you'd rather we invert it, I can do that, too. 👍

Reply:
Alright, good for me :)

But I think removing all other loggers would be a bit hard, sometimes people want file and syslog together? idk... but `logfile(..)` just adds the file logger, gues `syslog(..)` should behave the same?

Reply:
This method is really only used when you're trying to setup a logger to work a specific way (e.g. with a log file _or_ with syslog). I suppose someone may have a use case to do file and syslog together, but the current "flat" design doesn't readily support that. The problem is that there are usually default loggers added, but when I call the `syslog` method, I probably do not want those defaults (hence removing existing loggers first, just like the `logfile` method does).

If we refactored to an OO-design, then you will just define what loggers it should have via constructor args, and this method shouldn't generally be needed (except in the case where users are constantly changing around the default `logger`, but I don't know how common of a use case that is).

I think that trying to handle those types of multi-way logging scenarios with the current non-OO-design is going to make the code more complex, so my recommendation is we wait until the OO-design is in place and then we can support any arbitrary number of built-in loggers much more easily.

Thoughts?

Reply:
yeah, if we find the time we should move to a new design. hope we find the time, but there are so many priorities that i have the feeling it will be a while ^^